torchx/schedulers/ray_scheduler.py

```diff
@@ -132,7 +144,44 @@ def run_opts(self) -> runopts:
         return opts
 
     def schedule(self, dryrun_info: AppDryRunInfo[RayJob]) -> str:
-        raise NotImplementedError()
+        # TODO: What to do with cfg
+        cfg: RayJob = dryrun_info.request
+
+        # TODO: figure out how to use
+        # app_id not needed here - ray will return a job_id
+        app_id: str
+
+        # no cluser config is needed since users will call ray up
+        cluster_config_file: str
+
+        # can set this in driver script
+        cluster_name: Optional[str] = None
+        
+        # not sure what to do with this
+        verbose: bool = False
+
+        # not needed?
+        cfg.copy_script_dirs = True
+        cfg.copy_scripts = True
+
+        actors = cfg.actors
+        scripts = cfg.scripts
+
+        # Create serialized actors in ./ray/actors.json
+        serialize(actors)
+
+
+        job_id : str = self.client.submit_job(
+            # we will pack, hash, zip, upload, register working_dir in GCS of ray cluster
+            # and use it to configure your job execution.
+            # TODO: Make this work for windows
+            # TODO: How is ray/train.py called and pointed to 
+            # TODO: How is train.py called in DDP fashion
+            entrypoint="python ray/ray_driver.py",
+            runtime_env={"working_dir" : "./ray"}
```

Comment:
@amogkam my understanding is that there is no way to specify more than one "working_dir" in an environment definition, is that right? The problem I see right now is, since we can technically execute multiple user scripts that can be potentially in different paths on the local file system, ideally we should be able to specify a "working_dir" per actor. Looks like this is not possible right now. Similarly `ray_driver.py` is a separate file that is not in user's working directory. That needs to be copied separately as well. One idea I can think of is to copy the script directories into a temporary directory along with `ray_driver.py` and specify that temporary directory as "working_dir". It would work, but would also require unnecessarily copying things around and complicating the implementation. Any other ideas you can think of?

Reply:
@cbalioglu working_dir can also be configured at task and actor level, some of the examples can be seen in our tests: https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/tests/test_runtime_env_complicated.py?L822-825

For how currently runtime_env works in the path of jobs submission, we support two paths to minimize user copying files around that's already live on latest nightly wheels:
1) Upload user's local working_dir to running ray cluster with unique id, so all subsequent ray.init(), task / actor can use the same files with same working_dir from its runtime_env config.
2) Support a .zip file on s3 as working_dir in case user wants a stable set of files for repeating workload. This can work either as public bucket or private one with credentials set on the host that executes it.

Reply:
Yeah for now I think it might be best to package everything in a single working dir. 

So either 
1. Allow the user to only specify a single directory and expect all files to be in that directory. You would then just need to copy `ray_driver.py` into that directory (probably as a hidden file) and then specify that as the working directory
2. Or go with your suggestion of creating a new tmp directory by copying over all of the user's provided scripts and `ray_driver.py`.

It should be fine to use the same working_dir for every actor since the actual script that needs to be run is specified in the actor config.